.github/workflows/upload-e2e-test-reports.yml

```diff
@@ -0,0 +1,43 @@
+name: Upload reports
+
+on:
+  workflow_call:
+    outputs:
+      result_url:
+        description: "URL to see e2e test results"
+        value: ${{ jobs.upload-reports.outputs.result_url }}
+
+concurrency:
+  group: ${{ github.workflow }}
```

Comment:
this is what causes the action to queue if one is already running. we do this with our deploy as well, where its more common to see deploys queue when multiple pushes happen to the same branch